### PR TITLE
Linear scrambled halton

### PR DESCRIPTION
**Pull request by aaditj1962161**

### CODE REVIEW

File: qmcpy/discrete_distribution/halton.py [hunk 1]
```diff
@@ -88,11 +137,43 @@ def __init__(self, dimension=1, randomize=True, generalize=True, seed=None):
...
...6143, 6151, 6163, 6173, 6197, 6199, 6203, 6211, 6217, 6221, 6229, 6247, 6257, 6263, 6269, 6271, 6277, 6287, 6299, 6301, 6311, 6317, 6323, 6329, 6337, 6343, 6353, 6359, 6361, 6367, 6373, 6379, 6389, 6397, 6421, 6427, 6449, 6451, 6469, 6473, 6481, 6491, 6521, 6529, 6547, 6551, 6553, 6563, 6569, 6571, 6577, 6581, 6599, 6607, 6619, 6637, 6653, 6659, 6661, 6673, 6679, 6689, 6691, 6701, 6703, 6709, 6719, 6733, 6737, 6761, 6763, 6779, 6781, 6791, 6793, 6803, 6823, 6827, 6829, 6833, 6841, 6857, 6863, 6869, 6871, 6883, 6899, 6907, 6911, 6917, 6947, 6949, 6959, 6961, 6967, 6971, 6977, 6983, 6991, 6997, 7001, 7013, 7019, 7027, 7039, 7043, 7057, 7069, 7079, 7103, 7109, 7121, 7127, 7129, 7151, 7159, 7177, 7187, 7193, 7207, 7211, 7213, 7219, 7229, 7237, 7243, 7247, 7253, 7283, 7297, 7307, 7309, 7321, 7331, 7333, 7349, 7351, 7369, 7393, 7411, 7417, 7433, 7451, 7457, 7459, 7477, 7481, 7487, 7489, 7499, 7507, 7517, 7523, 7529, 7537, 7541, 7547, 7549, 7559, 7561, 7573, 7577, 7583, 7589, 7591, 7603, 7607, 7621, 7639, 7643, 7649, 7669, 7673, 7681, 7687, 7691, 7699, 7703, 7717, 7723, 7727, 7741, 7753, 7757, 7759, 7789, 7793, 7817, 7823, 7829, 7841, 7853, 7867, 7873, 7877, 7879, 7883, 7901, 7907, 7919],
```
Comment (alegresor) [inline qmcpy/discrete_distribution/halton.py:145]: Maybe there is a way to not duplicate this long list.

File: qmcpy/discrete_distribution/halton.py [hunk 2]
```diff
@@ -88,11 +137,43 @@ def __init__(self, dimension=1, randomize=True, generalize=True, seed=None):
...
```
Comment (alegresor) [inline qmcpy/discrete_distribution/halton.py:155]: same with this one

File: qmcpy/discrete_distribution/halton.py [hunk 3]
```diff
@@ -4,6 +4,7 @@
 from numpy import *
 from .c_lib import c_lib
 import ctypes
+from math import *
```
Comment (alegresor) [inline qmcpy/discrete_distribution/halton.py:7]: what is this used for?

**Review by alegresor — CHANGES_REQUESTED** [review]
Looks great @aaditj1962161! Just some minor suggestions below. I didn't look to closely at the notebook, but we can always change that in the develop branch. In the notebook you may want a to do a little speed comparison just to show that LMS with digital shift is a good bit slower than the other randomization techniques.

File: qmcpy/discrete_distribution/halton.py (hunk 3 above)
Comment (aaditj1962161) [reply qmcpy/discrete_distribution/halton.py:7]: It is used for the ceiling function

Comment (alegresor) [reply qmcpy/discrete_distribution/halton.py:7]: Sounds good. `numpy.ceil` works if you need it vectorized, but either way is fine.

File: qmcpy/discrete_distribution/halton.py (hunk 1 above)
Comment (aaditj1962161) [reply qmcpy/discrete_distribution/halton.py]: I have implemented a way in the halton class. is that fine?

**Review by alegresor — APPROVED** [review]
Excellent @aaditj1962161 ! Go ahead and merge when you are ready.

**Pull request merged**